Birth by Sleep Tools

Truthkey opened the pull request:
This PR includes a series of tools to edit parameters and events from Birth by Sleep.

EPD Editor:
> Allows you edit enemy parameters, such as Max Health, EXP multiplier or status ailments.

BEP Editor:
> Allows you to change base enemy statistics by world level.

BBS Event Editor:
> Made some improvements to the way it handles data.

ITB Editor:
> Allows you to edit or add information related to contents in treasure chests.

ITC Editor:
> Allows you to edit the reward obtained by collectibles (Stickers).

ITE Editor:
> Allows you to preview the list of valid item indices used by the game.

Xeeynamo commented:
## BEP EDITOR

Could you put in two separate tabs Base Stats and Disappear param? It would also be cool if the documentation could refer that `EnemyCommon.bep` is found in `arc/system/commongame.arc`. Can we have a short description (in the docs) of what Disappear Parameters do, if possible?

![image](https://user-images.githubusercontent.com/6128729/108590268-6ee9d080-735a-11eb-8e11-624141d2fcb4.png)

## EPD EDITOR

The editor looks all right, but I was not able to locate any EPD in my BBS copy. Where can I find those files?

## ITB EDITOR

The editor is a bit slow, but it does its job! In the docs there is no reference that this can be found in the ITEM folder.

## ITC EDITOR

Interestingly, I was not able to find this file in my BBSJP copy. The tool looks all right though.

## ITE EDITOR

👌

Xeeynamo commented on OpenKh.Tools.BbsEventTableEditor/ViewModels/EventViewModel.cs:
```diff
@@ -61,15 +86,26 @@ public byte Room
                 Event.Room = (byte)Math.Min(99, (int)value);
                 OnPropertyChanged(nameof(Name));
                 OnPropertyChanged(nameof(MapPath));
+                OnPropertyChanged(nameof(Rooms));
```
**BUG**: You might want to add `OnPropertyChanged(nameof(Rooms));` when the property `World` changes too or the list of rooms will not change.

Truthkey replied:
Thanks. I have added this change.

Truthkey commented:
> ## BEP EDITOR
> Could you put in two separate tabs Base Stats and Disappear param? It would also be cool if the documentation could refer that `EnemyCommon.bep` is found in `arc/system/commongame.arc`. Can we have a short description (in the docs) of what Disappear Parameters do, if possible?
> 
> ![image](https://user-images.githubusercontent.com/6128729/108590268-6ee9d080-735a-11eb-8e11-624141d2fcb4.png)
> 
> ## EPD EDITOR
> The editor looks all right, but I was not able to locate any EPD in my BBS copy. Where can I find those files?
> 
> ## ITB EDITOR
> The editor is a bit slow, but it does its job! In the docs there is no reference that this can be found in the ITEM folder.
> 
> ## ITC EDITOR
> Interestingly, I was not able to find this file in my BBSJP copy. The tool looks all right though.
> 
> ## ITE EDITOR
> 👌

I have changed all the things you've requested. Changes to documentation will be made in a separate PR, but I've already added such documentation to that PR.

As for EPD files go, they are included within any .`.arc` file within `arc/boss` or `arc/enemy`.

ITC files can be found in either the `item` folder or in these three files: `arc/system/common_aq.arc`, `arc/system/common_ve.arc`, `arc/system/common_te.arc`.

Xeeynamo reviewed (approved):
all good 👍 remember my suggestion to add the location of those files in the openkh documentation, it will be very useful to newbies of BBS modding like me
